core/src/main/java/com/alibaba/nacos/core/code/SpringApplicationRunListener.java

```diff
@@ -40,14 +38,9 @@ public class SpringApplicationRunListener implements org.springframework.boot.Sp
     private final SpringApplication application;
     
     private final String[] args;
-    
-    private List<NacosApplicationListener> nacosApplicationListeners = new ArrayList<>();
-    
-    {
-        nacosApplicationListeners.add(new LoggingApplicationListener());
-        nacosApplicationListeners.add(new StartingApplicationListener());
-    }
-    
+
+    Collection<NacosApplicationListener> nacosApplicationListeners = NacosServiceLoader.load(NacosApplicationListener.class);
```

Comment:
What will happen when load throw excepetion?

Reply:
The situation is not considered, I will test it again

Reply:
When the spi load fail and throw error, nacos cannot start.

Reply:
but the error will be ClassNoDefined which is hard to find problem.

Reply:
I think this happens rarely, because the IDE will prompt an error during the development process.